### ISSUE TITLE
[Suggestion]: Add Mod Item Option to Delete and Unsubscribe

### ISSUE THREAD

**Issue by hk21702** [RimSort/RimSort#607]
### Origin of idea

Though deletion of local files and unsubscribing from a mod in steam workshop are distinct operations, some users have expressed confusion. This is exacerbated by the fact that unsubscribing via steam (both via RimSort and via Steam itself) leaves dds files behind.

### Suggestion/proposed solution?

A new option for steam workshop mods, when steam integration is enabled, to unsubscribe from a steam workshop mod *and* delete all local files for that mod.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Duplicate Issue Check

- [X] I've checked for duplicate issues

### Wiki/FAQ Check

- [X] I've checked the Wiki for a solution

**Comment by twstagg** [comment]
Copying my message from Discord:

> The inherent reason for this is because Steam is supposed to delete the mod on unsubscribe. However, when you use Steamworks API to unsubscribe from the mod, it will change your Steam user's subscription state to that PublishedFileID, but it does not trigger a file verification like it does in Steam client when you use it to unsubscribe to a mod. If you are a Steam user and you unsubscribe to mod(s) via workshop menu option in RimSort, it is recommended that you use Help -> Validate Steam Client Mods in RimSort. Otherwise, you may complete the same operation in the game's Properties in your Steam Library.
> 
> Rather than deal with the finicky situation of "What if we just make RimSort delete the mod manually after unsubscribing", we opt for the above instead, because doing will definitely cause unintended behavior from Steam client itself, which already has a lot of issues handling it's own status of it's installed mods / subscriptions without the complicated aspects of putting a mod manager into the mix of that.

I think that if such an option were to be implemented, the above should be considered.

**Comment by LionelColaso** [comment]
implemented
